### PR TITLE
feat(unlock-app)- convert to zod

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description
Convert type `Paywall` types definition with `zod` in order to the validation and JSON schema later from it.

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by kalidiagne** [comment]
Maybe we can move all to `/packages/types` and we can later remove all the types defined inside `/unlock-app`

File: unlock-app/src/unlockTypes.ts [hunk 1]
```diff
@@ -2,6 +2,67 @@
...
+export const MetadataInputSchema = z.object({
+  name: z.string(),
+  defaultValue: z.string().optional(),
+  type: z.enum(['text', 'date', 'color', 'email', 'url']),
+  required: z.boolean(),
+  placeholder: z.string().optional(),
+  public: z.boolean().optional(), // optional, all non-public fields are treated as protected
+})
+
+export const PaywallCallToActionSchema = z.object({
```
Comment (julien51) [inline unlock-app/src/unlockTypes.ts:16]: We don't support this anymore...

**Comment by julien51** [comment]
We need @searchableguy 's feedback here

File: unlock-app/src/unlockTypes.ts (hunk 1 above)
Comment (searchableguy) [reply unlock-app/src/unlockTypes.ts:16]: Agreed. Let's remove it but keep it pass through so it doesn't fail validation.

File: unlock-app/src/unlockTypes.ts [hunk 2]
```diff
@@ -2,6 +2,67 @@
...
+  noWallet: z.string(),
+  metadata: z.string(),
+  card: z.string(),
+  quantity: z.string(),
+})
+
+export const PaywallConfigLockSchema = z.object({
+  name: z.string().optional(),
+  network: z.number().int().positive().optional(),
+  metadataInputs: z.array(MetadataInputSchema),
```
Comment (searchableguy) [inline unlock-app/src/unlockTypes.ts:30]: Should be optional, no?

File: unlock-app/src/unlockTypes.ts [hunk 3]
```diff
@@ -2,6 +2,67 @@
...
+  recurringPayments: z.boolean().optional(),
```
Comment (searchableguy) [inline unlock-app/src/unlockTypes.ts:31]: It's not a boolean. It's a number.

**Review by searchableguy — APPROVED** [review]
LGTM

File: unlock-app/src/unlockTypes.ts [hunk 4]
```diff
@@ -2,6 +2,67 @@
...
+
+export const PaywallConfigSchema = z.object({
+  title: z.string().optional(),
+  icon: z.string().optional(),
+  callToAction: PaywallCallToActionSchema.partial().optional(),
+  locks: z.record(PaywallConfigLockSchema),
+  metadataInputs: z.array(MetadataInputSchema),
+  persistentCheckout: z.boolean().optional(),
+  redirectUri: z.boolean().optional(),
+  useDelegatedProvider: z.boolean().optional(),
```
[the remark this answers is not in the document]
Comment (julien51) [reply unlock-app/src/unlockTypes.ts:52]: Well, I was going to say yes, but then someone asked for this today! So I'd say let's keep it!

**Pull request merged**